src/lib/bootloader/finish_client.rb

```diff
@@ -43,7 +43,7 @@ def write
       end
 
       # we do not manage bootloader, so relax :)
-      return true if bl_current.name == "none"
+      return true if bl_current.name == "none" || bl_current.stage1.devices.empty?
```

Comment:
do you test it with EFI? EFI does not have stage1. In general, I am not sure if this check is good idea. Bug is about selecting "none", so deselection lower looks reasonable, but not this additional for empty stage1.